src/gretel_client/transformers/transformers/bucket.py

```diff
@@ -1,34 +1,50 @@
 from typing import Union, List, Optional, Tuple
 
 from gretel_client.transformers.base import TransformerConfig, Transformer
 
 
-class BucketError(Exception):
-    pass
-
-
-class BucketRange:
-    def __init__(self, buckets: List[Tuple], *, labels: Union[List[Number], List[str]] = None, min_label=None,
-                 max_label=None,
-                 auto_label_prefix='BUCKET_'):
-        # buckets should be all strings or all numbers
-        str_check = [isinstance(item[0], str) and isinstance(item[1], str) for item in buckets]
-        num_check = [isinstance(item[0], Number) and isinstance(item[1], Number) for item in buckets]
-        if not all(str_check) and not all(num_check):
-            raise BucketError('buckets must be all strings or all numbers')
-        self.buckets = buckets
-        if not labels:
-            labels = [auto_label_prefix + str(x) for x in range(len(buckets))]
-        self.labels = labels
-        self.min_bucket = min_label if min_label is not None else labels[0]
-        self.max_bucket = max_label if max_label is not None else labels[-1]
-
-
 @dataclass(frozen=True)
 class BucketConfig(TransformerConfig):
-    bucket_range: BucketRange = None
+    buckets: Union[List[float], List[int], Tuple[Union[float, int], Union[float, int], Union[float, int]]] = None
+    bucket_labels: Union[List[float], List[int], List[str]] = None
+    lower_outlier_label: Union[float, int, str] = None
+    upper_outlier_label: Union[float, int, str] = None
+
+    @staticmethod
+    def bucket_tuple_to_list(buckets: Tuple[Union[float, int], Union[float, int], Union[float, int]] = None):
+        if buckets is None:
+            raise ValueError(f"buckets must be a 3-Tuple (min, max, width)")
+        b = []
+        current_min = buckets[0]
+        while current_min < buckets[1]:
+            b.append(current_min)
+            current_min += buckets[2]
+        b.append(buckets[1])
+        return b
+
+    @staticmethod
+    def get_bucket_labels_from_tuple(
+            buckets: Tuple[Union[float, int], Union[float, int], Union[float, int]] = None,
+            method: str = None) \
+            -> Union[List[float], List[int]]:
+        if buckets is None:
+            raise ValueError(f"buckets must be a 3-Tuple (min, max, width)")
+        # Use average by default
+        method = method or "avg"
+        if method not in ["min", "max", "avg"]:
+            raise ValueError(f"method must be one of 'min', 'max', 'avg': {method}")
+        bucket_labels = []
+        current_min = buckets[0]
+        while current_min < buckets[1]:
+            if method == "min":
+                bucket_labels.append(current_min)
+            elif method == "max":
+                bucket_labels.append(min(current_min + buckets[2], buckets[1]))
+            elif method == "avg":
+                bucket_labels.append(current_min + (buckets[2] / 2))
+            current_min += buckets[2]
+        return bucket_labels
```

Comment:
Ditto here. Module level function preferred plus docstrings w/ explains of what's going on and why. Also type hinting return value